beancount_import/matching.py

```diff
@@ -255,6 +295,86 @@ def add_posting(self, entry: Transaction, mp: MatchablePosting):
         group = self._postings.setdefault(_date_key(entry, mp), {})
         group[source_posting_ids] = (entry, mp)
 
+        weight = get_posting_weight(mp.posting)
+        if weight is None:
+            return
+
+        for dc in self.fuzz_date_currency_key(self._date_currency_key(entry, mp)):
+            self._date_currency[dc].append(self._search_posting(entry, mp))
+            self._date_currency_dirty[dc] = True
+
+    def get_date_currency_postings(self, key: DateCurrencyKey) -> List[SearchPosting]:
+        if self._date_currency_dirty[key]:
+            self._date_currency[key].sort()
```

Comment:
could save one dict lookup by setting:

results = self._date_currency[key] before the if statement

Reply:
This one is actually tricky. I use self._date_currency_dirty to tell me if I need to sort self._date_currency. They are different dicts.

Reply:
Ah, missed that, sorry.